packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -722,188 +572,156 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
+        // if the trader is selling WETH to us i.e if we are selling oSQTH
+        if (_order.traderToken == weth) {
+            IWETH9(weth).withdraw(IWETH9(weth).balanceOf(address(this)));
+            // if last param is false, transfer happens again
+            _mintWPowerPerp(_order.trader, _order.managerAmount, _order.traderAmount, true);
+            priceAtLastHedge = _order.traderAmount.mul(1e18).div(_order.managerAmount);
+        } else {
+            // oSQTH in, WETH out
+            _burnWPowerPerp(_order.trader, _order.traderAmount, _order.managerAmount, true);
+            //wrap it
+            IWETH9(weth).deposit{value: _order.managerAmount}();
+            // if last param is false, transfer happens again
+            priceAtLastHedge = _order.managerAmount.mul(1e18).div(_order.traderAmount);
         }
 
-        emit ExecuteBuyAuction(_seller, _wSqueethToBuy, _ethToSell, _isHedgingOnUniswap);
+        IERC20(_order.managerToken).transfer(_order.trader, _order.managerAmount);
+
+        emit HedgeOTC(
+            _order.trader, // market maker
+            _order.managerAmount, // token out
+            _order.traderAmount, // token in
+            sellerPrice
+        );
     }
 
     /**
-     * @notice determine auction direction, price, and ensure auction hasn't switched directions
-     * @param _auctionTriggerTime auction starting time
-     * @return auction type
-     * @return WSqueeth amount to sell or buy
-     * @return ETH to sell/buy
-     * @return auction WSqueeth/ETH price
+     * @dev hedge function to reduce delta using an array of signed orders
+     * @param managerSellAmount quantity the manager wants to sell
+     * @param managerBuyPrice the price at which the manager is buying
+     * @param _orders an array of signed order to swap tokens
      */
-    function _startAuction(uint256 _auctionTriggerTime)
-        internal
-        returns (
-            bool,
-            uint256,
-            uint256,
-            uint256
-        )
-    {
-        (uint256 strategyDebt, uint256 ethDelta) = _syncStrategyState();
-        uint256 currentWSqueethPrice = IOracle(oracle).getTwap(
-            ethWSqueethPool,
-            wPowerPerp,
-            weth,
-            hedgingTwapPeriod,
-            true
-        );
-        uint256 feeAdjustment = _calcFeeAdjustment();
-        (bool isSellingAuction, ) = _checkAuctionType(strategyDebt, ethDelta, currentWSqueethPrice, feeAdjustment);
-        uint256 auctionWSqueethEthPrice = _getAuctionPrice(_auctionTriggerTime, currentWSqueethPrice, isSellingAuction);
-        (bool isStillSellingAuction, uint256 wSqueethToAuction) = _checkAuctionType(
-            strategyDebt,
-            ethDelta,
-            auctionWSqueethEthPrice,
-            feeAdjustment
-        );
+    function hedgeOTC(
+        uint256 managerSellAmount,
+        uint256 managerBuyPrice,
+        Order[] memory _orders
+    ) external onlyOwner {
```

Comment:
do we want this to be a multisig or EOA? If EOA, we might want to have something other than owner, so some is under multisig and some under EOA.